backend/utils/threads_utils.py

```diff
@@ -56,8 +56,53 @@ async def add_message_to_thread_list(org_id, room_id, message_id, request: Threa
     thread_message["created_at"] = str(datetime.utcnow())
 
     message["threads"].insert(0, thread_message)
-   
+
     response = await update_message(org_id, message_id, message)
-    
+
     return response, thread_message
 
+
+async def update_message_thread(
+    org_id, room_id, message_id, thread_id, payload: Thread
+) -> dict[str, Any]:
+    """Updates a message document in the database.
+    Args:
+        org_id (str): The organization id where the message is being updated.
+        message_id (str): The id of the parent message whose thread message is to be edited.
+        thread_id (str): The id of the thread message to be edited
+        payload (dict[str, Any]): the new thread mesage
+    Returns:
+        dict[str, Any]: The response returned by DataStorage's update method.
+    """
+    message = await get_message(org_id, room_id, message_id)
+
+    if not message:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Message not found"
+        )
+
+    if message["sender_id"] != payload["sender_id"]:
```

Comment:
@Shopiley I think we have an implementation problem on this line. The sender of a parent message is not necessarily the author of a thread message in the threads list. We need to be checking `thread_message["sender_id"]` here rather than the `message["sender_id"]`.

Reply:
Oh that's true, but once again the issue of not being able to access the thread_id pops up

Reply:
we need to write a query that retrieves the thread message itself

Reply:
Done, pls review.